packages/loopback/lib/application.ts

```diff
@@ -18,20 +20,45 @@ export class Application extends Context {
       if (!ctor) {
         throw new Error(`The controller ${b.key} was not bound via .toClass()`);
       }
-
+      
       const ctorFactory = (req: http.ServerRequest, res: http.ServerResponse, operationName: string) => {
         const requestContext = new Context(this);
         requestContext.bind('controller.current.ctor').to(ctor);
         requestContext.bind('controller.current.operation').to(operationName);
         requestContext.bind('http.request').to(req);
         requestContext.bind('http.response').to(res);
-        return requestContext.get(b.key);
+        //[rashmi] TODO calling getAuthenticatedUser() here only temporary until MiddleWare design/implementation is ready
+        //the reason to call here is - this is the only place right now, where we have access to to request, app context and 
+        //also where AppController's constructor is called where we need to inject the 'user' 
+        return new Promise((resolve, reject) => {
+          const metadata = getAuthenticateMetadata(ctor.prototype, operationName);
+          var isAuthRequired;
+          if (metadata != null) {
+            isAuthRequired = true;
+          } else {
+            isAuthRequired = false;
+          }
+          this.getAuthenticatedUser(isAuthRequired, req).then((user) => {
+            this.bind('authentication.user').to(user);
+            requestContext.get(b.key).then((value) => {
+              resolve(value);
+            });
+          });
+        });
+        //end of Temp code until MiddleWare design/implementation is ready
       };
       const apiSpec = getApiSpec(ctor);
       router.controller(ctorFactory, apiSpec);
     });
   }
 
+  //[rashmi] TODO  getAuthenticatedUser function here only temporary until MiddleWare design/implementation is ready
+  public async getAuthenticatedUser(isAuthRequired: boolean, request: http.ServerRequest) : Promise<Object> {
+    const strategy = await this.get('authentication.strategy');
+    const user = await getAuthenticatedUser(isAuthRequired, request, strategy as Strategy);
```

Comment:
I think we should really move away from `any`-typed bindings and start using generics to force people to specify the type when using `ctx.get` directly.

```ts
const strategy = await this.get<Strategy>('authentication.strategy');
// the following should throw a compiler error
const strategy = await this.get('authentication.strategy');
// the following should report a tslint error about using "any"
const strategy = await this.get<any>('authentication.strategy');
```

My concern is that the current code looks perfectly legit and it's difficult to spot that `strategy` has `any` type. Compare the following:

```ts
// current version - looks legit, but strategy is "any"

const strategy = await this.get('authentication.strategy');

// proposed version when using "any" - it's immediatelly
// clear that a detailed review is needed, we may want 
// to justify why we are using "any" type here

// tslint:disable-next-line:no-any
const strategy = await this.get<any>('authentication.strategy');
```

@strongloop/loopback-next thoughts?